tokio-rustls/src/lib.rs

```diff
@@ -1,4 +1,35 @@
 //! Asynchronous TLS/SSL streams for Tokio using [Rustls](https://github.com/ctz/rustls).
+//!
+//! # Why do I need to call `poll_flush`?
+//!
+//! Most TLS implementations will have an internal buffer to improve throughput,
+//! and rustls is no exception.
+//!
+//! When we write data to `TlsStream`, we always write rustls buffer first,
+//! then take out rustls encrypted data packet, and write it to data channel (like TcpStream).
+//! When data channel is pending, some data may remain in rustls buffer.
+//! at this point we must call flush to write again it into data channel.
+//!
+//! `tokio-rustls` To keep it simple and correct, [TlsStream] will behave like `BufWriter`.
```

Comment:
I find this hard to understand. I feel like there are three steps here that should be plainly stated:

1. When or in what kind of situations this problem arises ("Does this apply to me")
2. Context on the underlying problem; there is some of this, but as written it's hard to relate to as a library user
3. What to do to solve the issue (that is, in what kind of situations should I be calling `poll_flush()` explicitly)

Reply:
Sorry, I think it is common sense that `BufWriter` needs to call `flush`. so I subconsciously did not give more explanation.